### PR TITLE
Support `resource` record type with an http:// or https:// URL

### PR DESCRIPTION
**Pull request by vbanos**

### CODE REVIEW

File: cdx_writer.py [hunk 1]
```diff
@@ -642,6 +648,11 @@ def dispatch_http(self, record):
                     'WARC-Profile').endswith('/revisit/server-not-modified'):
                 return None
             return RevisitHandler
+        elif record.type == 'resource' and record.url.startswith(('http://', 'https://')):
+            m = ResourceHandler.RE_RESPONSE_LINE.match(record.content[1])
+            if m and m.group(1) == '304':
+                return None
+            return ResourceHandler
```
Comment (kngenie) [inline cdx_writer.py]: these three lines appear unnecessary. resource record has no response line.

File: cdx_writer.py [hunk 2]
```diff
@@ -642,6 +648,11 @@ def dispatch_http(self, record):
                     'WARC-Profile').endswith('/revisit/server-not-modified'):
                 return None
             return RevisitHandler
+        elif record.type == 'resource' and record.url.startswith(('http://', 'https://')):
+            m = ResourceHandler.RE_RESPONSE_LINE.match(record.content[1])
+            if m and m.group(1) == '304':
+                return None
```
Comment (kngenie) [inline cdx_writer.py]: lines 652-653 appear unnecessary. resource record has no response line.

File: cdx_writer.py [hunk 3]
```diff
@@ -555,6 +553,14 @@ def aif_meta_tags(self):
 
         return ''.join(s) if s else None
 
+class ResourceHandler(ResponseHandler):
```
Comment (kngenie) [inline cdx_writer.py]: `ResourceHandler` should not extend `ResponseHandler` (it is for regular HTTP response records, `response` and `revisits`).

Comment (vbanos) [inline cdx_writer.py]: I changed it to extend the `RecordHandler`

**Pull request merged**